app/assets/javascripts/discourse/app/components/user-badge.gjs

```diff
@@ -0,0 +1,25 @@
+import Component from "@glimmer/component";
+import BadgeButton from "discourse/components/badge-button";
+
+export default class UserBadge extends Component {
+  get showGrantCount() {
+    return this.args.count > 1;
+  }
+
+  get badgeUrl() {
+    // NOTE: I tried using a link-to helper here but the queryParams mean it fails
+    let username = this.args.user?.username_lower || "";
+    username = username !== "" ? "?username=" + username : "";
+    return this.args.badge.url + username;
```

Comment:
here comes

🏌️‍♂️ THE GOLFER MAN ⛳ 

😄 


```suggestion
    let username = this.args.user?.username_lower;
    username = username ? `?username=${username}` : "";
    return this.args.badge.url + username;
```

Reply:
yes agreed :+1: